Simplified/NYPLAnnouncementBusinessLogic.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+
+private let announcementsFilename: String = "NYPLPresentedAnnouncementsList"
+
+class NYPLAnnouncementBusinessLogic {
+  static let shared = NYPLAnnouncementBusinessLogic()
+
+  private var presentedAnnouncements: Set<String> = Set<String>()
+    
+  init() {
+    restorePresentedAnnouncements()
+  }
+    
+  func presentAnnouncements(_ announcements: [Announcement]) {
+    for announcement in announcements {
+      if shouldPresentAnnouncement(id: announcement.id) {
+        DispatchQueue.main.async {
+          let vc = NYPLAnnouncementViewController(announcement: announcement)
+          NYPLRootTabBarController.shared()?.safelyPresentViewController(vc, animated: true, completion: nil)
+        }
+      }
+    }
+  }
+  
+  // MARK: - Read
+    
+  fileprivate func restorePresentedAnnouncements() {
+    guard let filePathURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first?.appendingPathComponent(announcementsFilename),
+      let filePathData = try? Data(contentsOf: filePathURL),
+      let unarchived = try? NSKeyedUnarchiver.unarchiveTopLevelObjectWithData(filePathData),
+      let presented = unarchived as? Set<String> else {
+        return
+    }
+    presentedAnnouncements = presented
+  }
+    
+  fileprivate func shouldPresentAnnouncement(id: String) -> Bool {
+    return !presentedAnnouncements.contains(id)
+  }
+  
+  // MARK: - Write
+    
+  /// Add presented announcement to storage
+  /// No guarantees are being made about whether this is called on the main thread or not
```

Comment:
hmm, not sure this belongs here... was this referred to a completion block maybe?

I'd just say "this class is not thread safe" as documentation for the class rather than each method